[native] Add optional Task queuing framework. (#25093)
prestodb/presto: presto-docs/src/main/sphinx/presto_cpp/properties.rst

CPU threshold in % above which the worker is considered overloaded in terms of
CPU use. Ignored if zero.

``worker-overloaded-cooldown-period-sec``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``
* **Default value:** ``5``

> Specifies how many seconds worker has to be not overloaded in terms of

Reviewer: ```suggestion
Specifies how many seconds worker has to be not overloaded (in terms of
```

> memory and CPU before we consider it really not overloaded.

Reviewer: ```suggestion
memory and CPU) before its status changes to not overloaded.
```

This is to prevent spiky fluctuation of the overloaded status.

``worker-overloaded-task-queuing-enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``boolean``
* **Default value:** ``false``

> If true, the worker will start queuing new Tasks when overloaded and

Reviewer: ```suggestion
If true, the worker starts queuing new tasks while overloaded, and
```

> starting them gradually when it stops being overloaded.

Reviewer: ```suggestion
starts them gradually when it stops being overloaded.
```
